### PR TITLE
Update ls to the latest spec

### PR DESCRIPTION
**Pull request by diasdavid**
- [x] update deps
- [x] update ls

### CODE REVIEW

**Comment by diasdavid** [comment]
Updated to the latest spec. I feel that the multistream API is really wonky, it was definitely built long time ago. I will branch this branch and come up with a new, cleaner API, without more intuitive naming for the calls.

File: examples/tcp-interactive-ls.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+'use strict'
+
+var tcp = require('net')
+var Interactive = require('../src').Interactive
```
Comment (dignifiedquire) [inline examples/tcp-interactive-ls.js]: Why not make them tests rather than examples, that way they stay up to date

File: package.json [hunk 2]
```diff
@@ -30,16 +30,18 @@
     "labs"
   ],
   "engines": {
-    "node": "^4.2.0"
+    "node": "^4.4.0"
```
Comment (dignifiedquire) [inline package.json]: Is there a need for this?

File: package.json [hunk 3]
```diff
@@ -30,16 +30,18 @@
...
   },
   "author": "David Dias <daviddias@ipfs.io>",
   "license": "MIT",
   "dependencies": {
+    "async": "^2.0.0-rc.4",
```
Comment (dignifiedquire) [inline package.json]: Let's try to not use async, reference: https://github.com/ipfs/js-ipfs/issues/214 (it's just gigantic, close to 1Mb minified)

File: src/lib/select.js [hunk 4]
```diff
@@ -44,7 +45,22 @@ function Select () {
...
         if (msg === 'ls') {
-          lpm.write(duplexStream, new Buffer(JSON.stringify(Object.keys(handlers)) + '\n'))
+          var protos = Object.keys(handlers)
+          var nProtos = protos.length
+          let size = 0 // total size of the list of protocols, including varint and newline
+          protos.forEach((proto) => {
+            var p = new Buffer(proto + '\n')
+            var el = varint.encodingLength(p.length)
+            size += el
+          })
```
Comment (dignifiedquire) [inline src/lib/select.js]: this would be much nicer using `reduce`

**Comment by diasdavid** [comment]
Closing this one in favour of https://github.com/diasdavid/js-multistream/pull/13

File: src/lib/select.js (hunk 4 above)
Comment (diasdavid) [inline src/lib/select.js]: good point. changed on #13

File: package.json (hunk 3 above)
Comment (diasdavid) [inline package.json]: Sounds good. Should I bake my own or do you have a recommendation for `async.times` replacement that is really tiny?

Comment (dignifiedquire) [inline package.json]: I would probably just do sth like this.

```js
const parallel = require('run-parallel')
const range = require('lodash.range')

// before
async.times(10, (i, cb) => cb(null, i * i), (err, res) => console.log(res))

// after
const times = (i, work, cb) => {
  parallel(range(i).map((i) => (cb) => work(i, cb)), cb)
}
times(10, (i, cb) => cb(null, i * i), (err, res) => console.log(res))
```

**Pull request closed without merging**